pymc/sampling.py

```diff
@@ -253,6 +255,38 @@ def all_continuous(vars):
         return True
 
 
+def _get_seeds_per_chain(
+    random_state: RandomState,
+    chains: int,
+) -> Sequence[int]:
+    """Obtain new or validate specified integer seeds per chain"""
+
+    def _get_unique_seeds_per_chain(integers_fn):
+        seeds = []
+        while len(set(seeds)) != chains:
+            seeds = [int(seed) for seed in integers_fn(2**30, dtype=np.int64, size=chains)]
+        return seeds
```

Comment:
We could maybe avoid drawing as many integers as the total number of `chains` every time? Maybe we could use an order preserving set like entity? Something like a dictionary with no value corresponding to the keys?
```suggestion
        seeds = {}
        while len(set(seeds)) < chains:
            seeds.update(
                {int(seed): None for seed in integers_fn(2**30, dtype=np.int64, size=chains - len(seeds))}
            )
        return list(seeds)
```

Reply:
I am not too concerned with this. With 10_000 chains we would still expect zero collisions 95% of the time, and with 1000 chains only 99.95% of the time.

I would rather leave it in this simple form.  This logic will be entirely removed once we stop using global seeding for our PyMC samplers.